### PR TITLE
Updates Hub Db deployment in case of Openshift

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

-   This patch adds a transformer which updates the db deployment,
    in case of openshift to use productized postgres images and also
    replace a few env and commands for readiness and liveness probes
    
 Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: pkg/reconciler/openshift/tektonhub/extension.go
Comment (sm43) [inline pkg/reconciler/openshift/tektonhub/extension.go:261]: this could be simplied as as
```
_, ok := map[e.name]
if ok {
envs[i].Name = map[e.name]
}
```
and the map would be
```
replaceVal := map[string]string {
"POSTGRES_DB": "POSTGRESQL_DATABASE"
}
```

File: pkg/reconciler/openshift/tektonhub/extension.go
Comment (sm43) [inline pkg/reconciler/openshift/tektonhub/extension.go:205]: add unit test pls

**Review by vdemeester — APPROVED** [review]
/hold
We need the two dependent PR to get in before this one, but LGTM

File: pkg/reconciler/openshift/tektonhub/extension.go
Comment (nikhil-thomas) [inline pkg/reconciler/openshift/tektonhub/extension.go:150]:
```suggestion
        routeManifest := uiManifest.Filter(mf.ByKind("Route"))
	routeManifest, err = routeManifest.Transform(
		mf.InjectOwner(th),
		mf.InjectNamespace(targetNs),
	)
	if err != nil {
		logger.Error("failed to transform manifest")
		return err
	}

	if err := routeManifest.Apply(); err != nil {
		return err
	}
```

how about this, as we are only creating the routes in here

Comment (nikhil-thomas) [reply pkg/reconciler/openshift/tektonhub/extension.go:150]: same applies to lines 95 for API route

**Comment by nikhil-thomas** [comment]
@PuneetPunamiya @piyush-garg should the route creation for ui be handled in the post-reconcile instead of pre-reconcile.

**Comment by PuneetPunamiya** [comment]
> @PuneetPunamiya @piyush-garg should the route creation for ui be handled in the post-reconcile instead of pre-reconcile.

@nikhil-thomas  The reason we are creating the the ui route in the pre-reconciler is because the config map of ui will require the ui route, 

For this one option could be
- We create the route in `manageUiComponent` function before creating the configMap for ui, 
- The other alternative would be we can add a field in hub cr as `ui: true` which means the user wants ui as well and in the pre-reconciler we can check the value of `ui` and based on that value we create the ui route, but then we'll have to add a check in the reconciler as well to not go forward to create the ui component

**Pull request merged**
